src/tools/rbd/action/Export.cc

```diff
@@ -116,34 +386,141 @@ static int do_export(librbd::Image& image, const char *path, bool no_progress)
   }
 
   utils::ProgressContext pc("Exporting image", no_progress);
-
   SimpleThrottle throttle(max_concurrent_ops, false);
   uint64_t period = image.get_stripe_count() * (1ull << info.order);
-  for (uint64_t offset = 0; offset < info.size; offset += period) {
-    if (throttle.pending_error()) {
-      break;
+
+  if (export_format == 2) {
```

Comment:
Minor: since format 1 and 2 are so different, it would be cleaner to separate the implementation bodies into a v1 and v2 export method